pkg/txlist_validator/tx_list_validator.go

```diff
@@ -29,24 +30,38 @@ func NewTxListValidator(
 }
 
 // ValidateTxList checks whether the transactions list in the TaikoL1.proposeBlock transaction's
-// input data is valid.
+// input data is valid, the rules are:
+// - If the transaction list is empty, it's valid.
+// - If the transaction list is not empty:
+//  1. If the transaction list is using calldata, the compressed bytes of the transaction list must be
+//     less than or equal to maxBytesPerTxList.
+//  2. The transaction list bytes must be able to be RLP decoded into a list of transactions.
 func (v *TxListValidator) ValidateTxList(
 	blockID *big.Int,
 	txListBytes []byte,
 	blobUsed bool,
-) (isValid bool) {
+) bool {
 	// If the transaction list is empty, it's valid.
 	if len(txListBytes) == 0 {
 		return true
 	}
 
 	if !blobUsed && (len(txListBytes) > int(v.maxBytesPerTxList)) {
-		log.Info("Transactions list binary too large", "length", len(txListBytes), "blockID", blockID)
+		log.Info("Compressed transactions list binary too large", "length", len(txListBytes), "blockID", blockID)
 		return false
 	}
 
-	var txs types.Transactions
-	if err := rlp.DecodeBytes(txListBytes, &txs); err != nil {
+	var (
+		txs types.Transactions
+		err error
+	)
+
+	if txListBytes, err = utils.Decompress(txListBytes); err != nil {
+		log.Info("Failed to decompress tx list bytes", "blockID", blockID, "error", err)
+		return false
+	}
+
+	if err = rlp.DecodeBytes(txListBytes, &txs); err != nil {
```

Comment:
here decompress & decode are only tentative operation?? Can it return final txs to save dup decompress & decode later on (I guess it's there, not sure).